[GEOS-9995]: GWC-MBTiles community module (#4928)
geoserver/geoserver: doc/en/user/source/community/gwc-mbtiles/index.rst

.. _community_gwc_mbtiles:

GWC MBTiles layer plugin
========================

> This plugin allows to adding a GWC MBTiles Layer to GeoServer's cached Tile Layers configuration so that GeoServer can load cached tiles from ready-to-use MBTiles.

Reviewer: ```suggestion
This plugin allows adding a GWC MBTiles Layer to GeoServer's cached Tile Layers configuration so that GeoServer can load cached tiles from ready-to-use MBTiles.
```

Installation
------------

> As a community module, make user you have donwloaded the :download_extension:`gwc-mbtiles`

Reviewer: ```suggestion
As a community module, make sure you have downloaded the :download_extension:`gwc-mbtiles`
```

Author: Ah ah. I have copied that line from another module already using the download_extension MACRO.
Didn't notice that :).
Fixing it.

To install the module, unpack the zip file contents into the GeoServer ``WEB-INF/lib`` directory and restart GeoServer.

Configuring a MBTilesLayer
--------------------------
The configuration of a new MBTilesLayer can be done by editing the geowebcache configuration file located in :file:`<data_dir>/gwc/geowebcache.xml`

Locate the `layers` section of the config or add it if missing.

Then define a new `mbTilesLayer` node for each new MBTilesLayer you want to add. A GWC configuration containing a configured MBTiles Layer would look like this:

.. code-block:: xml

   <gwcConfiguration>
     <!-- ... -->
     <layers>

>        <!-- ... -->

Reviewer 2: A minimal but valid configuration for the layer would be helpful. GeoServer users are not used to play with GWC config files, giving them a "copy, paste and customize" approach is going to reduce confusion and questions.

>        <!-- ... -->

Reviewer 2: ```suggestion
       <!-- Other layer definitions may be here -->
```

         <mbtilesLayer>
           <tilesPath>D:\Data\mbtiles\countries-raster.mbtiles</tilesPath>
           <tileSize>256</tileSize>
           <name>countries</name>
           <metaInformation>
             <title>Countries</title>
             <description>Raster Layer with Countries</description>
           </metaInformation>
         </mbtilesLayer>

>        <!-- ... -->

Reviewer 2: ```suggestion
       <!-- Other layer definitions may be here -->
```

     </layers>
   </gwcConfiguration>

A few note on the above configuration elements of an mbtilesLayer definition:

* `tilesPath` (mandatory) is the path to the MBTiles file containing the tiles.
* `tileSize` is the size of the tiles stored on the MBTiles file. When not set, it will be defaulted to 256.

> * `name` (optional) represents the name to be assigned to the layer. If not specified, the name field of the metadata table stored in the MBTiles file will be used.

Reviewer 2: What if the name field is also missing?

Author: I have updated the doc with a clarification

.. _community_gwc_mbtiles:

GWC MBTiles layer plugin
========================

This plugin allows to add GWC MBTiles Layer to GeoServer cached Tile Layers configuration so that it can load cached tiles from ready-to-use MBTiles.

Installation
------------

> As a community module, the package needs to be downloaded from the `nightly builds <https://build.geoserver.org/geoserver/>`_,

Reviewer: You might want to consider using the `:download_extension:` macro.

picking the community folder of the corresponding GeoServer series (e.g. if working on the GeoServer main development branch nightly
builds, pick the zip file form ``main/community-latest``).

To install the module, unpack the zip file contents into the GeoServer ``WEB-INF/lib`` directory and restart GeoServer.

Configuring a MBTilesLayer
--------------------------
The configuration of a new MBTilesLayer can be done by editing the geowebcache configuration file located in :file:`<data_dir>/gwc/geowebcache.xml`

Locate the `layers` section of the config or add it if missing.

> Then define a new `mbTilesLayer` node for each new MBTilesLayer you want to add. A GWC Configuration containing a configured MBTiles Layer would look like this:

Reviewer: ```suggestion
Then define a new `mbTilesLayer` node for each new MBTilesLayer you want to add. A GWC configuration containing a configured MBTiles Layer would look like this:
```

.. code-block:: xml

   <gwcConfiguration>
     <!-- ... -->
     <layers>
       <!-- ... -->
         <mbtilesLayer>
           <tilesPath>D:\Data\mbtiles\countries-raster.mbtiles</tilesPath>
           <tileSize>256</tileSize>
           <name>countries</name>
           <metaInformation>
             <title>Countries</title>
             <description>Raster Layer with Countries</description>
           </metaInformation>
         </mbtilesLayer>
       <!-- ... -->
     </layers>
   </gwcConfiguration>

A few note on the above configuration elements of an mbtilesLayer definition:

* `tilesPath` (mandatory) is the path to the MBTiles file containing the tiles.
* `tileSize` is the size of the tiles stored on the MBTiles file. When not set, it will be defaulted to 256.

> * `name` (optional) represents the name to be assigned to the layer. If not specified, the name field of the metadata table stored on the mbtiles will be used.

Reviewer: ```suggestion
* `name` (optional) represents the name to be assigned to the layer. If not specified, the name field of the metadata table stored in the MBTiles file will be used.
```

> * `metaInformation` with `title` and `description` are optionals tags. They will be exposed in the capabilities document when available.

Reviewer: ```suggestion
* `metaInformation` with `title` and `description` are optional tags. They will be exposed in the capabilities document when available.
```
